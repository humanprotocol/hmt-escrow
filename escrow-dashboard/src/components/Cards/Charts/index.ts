import CardBarChart from './BarChart';
import CardStackedBarChart from './StackedBarChart';

export { CardBarChart, CardStackedBarChart };
