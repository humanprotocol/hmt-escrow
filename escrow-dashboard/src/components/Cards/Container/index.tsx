import Box from '@mui/material/Box';
import React from 'react';

export const CardContainer = ({ children }: { children: React.ReactNode }) => {
  return (
    <Box
      sx={{
        background: '#fff',
        boxShadow:
          '0px 3px 1px -2px #E9EBFA, 0px 2px 2px rgba(233, 235, 250, 0.5), 0px 1px 5px rgba(233, 235, 250, 0.2)',
        borderRadius: { xs: '8px', xl: '16px' },
        padding: { xs: '21px 24px 12px', xl: '54px 60px 32px' },
      }}
    >
      {children}
    </Box>
  );
};
