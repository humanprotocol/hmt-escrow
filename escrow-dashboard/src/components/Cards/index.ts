export * from './Charts';
export { CardTextBlock } from './TextBlock';
