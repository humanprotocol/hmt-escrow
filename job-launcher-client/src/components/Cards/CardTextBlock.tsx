import * as React from 'react';
import { Box, CircularProgress, Divider, Typography } from '@mui/material';

interface ICardTextBlock {
  title?: string;
  value?: number | string;
}

export const CardTextBlock: React.FC<ICardTextBlock> = ({
  title = '',
  value,
}): React.ReactElement => (
  <>
    <Divider textAlign="center">
      <Typography variant="body2" color="text.secondary">
        {title}
      </Typography>
    </Divider>
    {value !== undefined ? (
      <Typography variant="body2" textAlign="center">
        {Number.isNaN(value) ? <>&nbsp;</> : value}
      </Typography>
    ) : (
      <Box display="flex" justifyContent="center">
        <CircularProgress size="1.5rem" />
      </Box>
    )}
  </>
);
