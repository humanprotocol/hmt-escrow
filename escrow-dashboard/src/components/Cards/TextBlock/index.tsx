import { Typography } from '@mui/material';
import numeral from 'numeral';
import React from 'react';

import { CardContainer } from '../Container';

type CardTextBlockProps = {
  title: string;
  value?: number | string;
  format?: string;
};

export const CardTextBlock: React.FC<CardTextBlockProps> = ({
  title,
  value,
  format = '0,0',
}): React.ReactElement => {
  return (
    <CardContainer>
      <Typography variant="body2" color="primary" fontWeight={600} mb="4px">
        {title}
      </Typography>
      <Typography
        variant="h2"
        color="primary"
        sx={{ fontSize: { xs: 32, md: 48, lg: 64, xl: 80 } }}
      >
        {numeral(value).format(format)}
      </Typography>
    </CardContainer>
  );
};
