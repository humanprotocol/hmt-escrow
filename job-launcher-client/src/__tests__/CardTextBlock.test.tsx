import React from "react";
import { render, screen } from "@testing-library/react";
import { act, create } from "react-test-renderer";
import { CardTextBlock } from "../components/Cards/CardTextBlock";

const mock = {
  title: "Title",
  value: 10,
};

describe("when rendered with `title` prop", () => {
  it("should paste it into the component", () => {
    render(<CardTextBlock title={mock.title} value={mock.value} />);
    expect(screen.getByText(mock.title)).toBeInTheDocument();
  });
});

describe("when rendered with `value` prop", () => {
  it("should paste it into the component", () => {
    render(<CardTextBlock title={mock.title} value={mock.value} />);
    expect(screen.getByText(mock.value)).toBeInTheDocument();
  });
});

describe("when rendered without `value` prop", () => {
  it("should show the loading spinner in the component", () => {
    render(<CardTextBlock title={mock.title} />);
    expect(screen.getByRole("progressbar")).toBeInTheDocument();
  });
});

it("renders correctly", () => {
  let root;
  act(() => {
    root = create(<CardTextBlock title={mock.title} value={mock.value} />);
  });
  expect(root).toMatchSnapshot();
});
