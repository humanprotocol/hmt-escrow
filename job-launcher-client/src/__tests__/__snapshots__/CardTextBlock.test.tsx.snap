// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <div
    className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1ws8119-MuiDivider-root"
    role="separator"
  >
    <span
      className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Title
      </p>
    </span>
  </div>,
  <p
    className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
  >
    10
  </p>,
]
`;
